sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsServiceSerializer.cs

```diff
@@ -196,26 +196,29 @@ private static string ReadModelVersion(JsonElement documentElement)
 
         private static TextDocumentBatchStatistics ReadDocumentBatchStatistics(JsonElement documentElement)
         {
-            if (documentElement.TryGetProperty("statistics", out JsonElement statisticsElement))
+            if (!documentElement.TryGetProperty("statistics", out JsonElement statisticsElement))
             {
-                int documentCount = default;
-                int validDocumentCount = default;
-                int invalidDocumentCount = default;
-                long transactionCount = default;
+                return default;
+            }
 
-                if (statisticsElement.TryGetProperty("documentsCount", out JsonElement documentCountValue))
-                    documentCount = documentCountValue.GetInt32();
-                if (statisticsElement.TryGetProperty("validDocumentsCount", out JsonElement validDocumentCountValue))
-                    validDocumentCount = validDocumentCountValue.GetInt32();
-                if (statisticsElement.TryGetProperty("erroneousDocumentsCount", out JsonElement erroneousDocumentCountValue))
-                    invalidDocumentCount = erroneousDocumentCountValue.GetInt32();
-                if (statisticsElement.TryGetProperty("transactionsCount", out JsonElement transactionCountValue))
-                    transactionCount = transactionCountValue.GetInt64();
+            var documentDict = new Dictionary<string, long>()
+            {
+                {"documentsCount", default},
+                {"validDocumentCount", default},
+                {"invalidDocumentCount", default},
+                {"transactionCount", default}
+            };
 
-                return new TextDocumentBatchStatistics(documentCount, validDocumentCount, invalidDocumentCount, transactionCount);
+            foreach (var document in documentDict.ToList()) // ToList() purpose is to avoid error "Collection was modified"
+            {
+                if (statisticsElement.TryGetProperty(document.Key, out JsonElement countValue))
+                    documentDict[document.Key] = countValue.GetInt64();
```

Comment:
what would happen if in the future we the `statisticsElement` contains elements that are not numeric? i.e. it returns text?

Reply:
umm, i think it can be done using like this

```
var element = null;
if (statisticsElement.TryGetProperty(document.Key, out JsonElement countValue)) {
     if (countValue.ValueKind == System.Text.Json.JsonValueKind.String)
     {
           element = countValue.GetString();
     }
}

```

thank you for your review!